scripts/firedrake-install

```diff
@@ -212,6 +212,28 @@ def resolve_doi_branches(doi):
     return branches
 
 
+def honour_petsc_dir_get_petsc_dir():
+    try:
+        petsc_dir = os.environ["PETSC_DIR"]
+    except KeyError:
+        raise InstallError("Unable to find installed PETSc (did you forget to set PETSC_DIR?)")
+    petsc_arch = os.environ.get("PETSC_ARCH", "")
+    return petsc_dir, petsc_arch
+
+
+def honour_petsc_dir_get_petsc_int_type():
```

Comment:
This feels like we are duplicating a lot of logic from `get_petsc_packages()` or `get_petsc_config()`, could we just make one of these more generic, rather than duplicating code?

Reply:
Yes. Fixed!